### PR TITLE
[ci] Skip FPGA , SW test for CDC changes

### PR DESCRIPTION
**Pull request by eunchan**
This commit let CI skip FPGA tests, SW tests when only CDC related
changes are made.

### CODE REVIEW

**Review by tjaychen — APPROVED** [review]
o this looks good to me.

File: ci/scripts/get-build-type.sh [hunk 1]
```diff
@@ -57,7 +58,18 @@ if [[ "$build_reason" = "PullRequest" ]]; then
     else
         echo "PR doesn't contain OTBN changes"
     fi
+
+    # Check if the commit has only CDC related changes (run-cdc.tcl,
+    # cdc_waivers*.tcl).
+    echo "Checking for changes in this PR other than to CDC waivers"
+    git diff --quite "$merge_base" -- ':(attr:!cdc)*.tcl' && only_cdc_change=1 || true
```
Comment (milesdai) [inline ci/scripts/get-build-type.sh:65]: `--quite` --> `--quiet`

**Review by milesdai — APPROVED** [review]
Small typo but otherwise LGTM.

It's a bit hard for me to tell just from reading this if it's going to work 100% as expected, so it'd be good to keep an eye on this for a few days after merging to make sure that we're not accidentally testing more or less than what's expected.

File: ci/scripts/get-build-type.sh (hunk 1 above)
Comment (eunchan) [reply ci/scripts/get-build-type.sh:65]: oh. I thought I fixed it hehe

File: ci/scripts/get-build-type.sh [hunk 2]
```diff
@@ -57,7 +58,18 @@ if [[ "$build_reason" = "PullRequest" ]]; then
     else
         echo "PR doesn't contain OTBN changes"
     fi
+
+    # Check if the commit has only CDC related changes (run-cdc.tcl,
+    # cdc_waivers*.tcl).
+    echo "Checking for changes in this PR other than to CDC waivers"
+    git diff --quiet "$merge_base" -- ':(attr:!cdc)*.tcl' && only_cdc_change=1 || true
```
Comment (milesdai) [inline ci/scripts/get-build-type.sh:65]: Taking another look at this, I think the `':(attr:!cdc)*.tcl'` check might be too specific. This is going to check for changes in all files that do not have `cdc` specified _and_ end in `.tcl`. Instead, we want to check for changes in all files that do not have `cdc` specified (regardless of their file extension).

**Comment by eunchan** [comment]
Ah.. you are right! I should not have `*.tcl` rather `:(attr:!cdc)` thanks @milesdai !

**Pull request merged**